packages/app/src/containers/ColorSwatchesInspector.tsx

```diff
@@ -0,0 +1,302 @@
+        color={rgbaColor}
+        onChange={handleChange}
+      >
+        <Saturation />
+        <Spacer.Vertical size={12} />
+        <Hue />
+        <Spacer.Vertical size={5} />
+        <Alpha />
+      </ColorPicker>
+    </FullContent>
+  );
+});
+
+const ColorSelectRow = memo(function ColorSelectRow({
+  id,
+  name = "New Color Variable",
+  color,
+  hexValue = 'FFFFFF',
+  onChangeColor,
+  onChangeOpacity,
+  onNudgeOpacity,
+  onInputChange
+}: Props) {
+  const colorInputId = `${id}-color`;
+  const hexInputId = `${id}-hex`;
+  const opacityInputId = `${id}-opacity`;
+
+  const renderLabel = useCallback(
+    ({ id }) => {
+      switch (id) {
+        case colorInputId:
+          return <Label.Label>Color</Label.Label>;
+        case hexInputId:
+          return <Label.Label>Hex</Label.Label>;
+        case opacityInputId:
+          return <Label.Label>Opacity</Label.Label>;
+        default:
+          return null;
+      }
+    },
+    [colorInputId, hexInputId, opacityInputId],
+  );
+
+  const handleSubmitOpacity = useCallback(
+    (opacity: number) => {
+      onChangeOpacity(opacity / 100);
+    },
+    [onChangeOpacity],
+  );
+
+  const handleNudgeOpacity = useCallback(
+    (amount: number) => {
+      onNudgeOpacity(amount / 100);
+    },
+    [onNudgeOpacity],
+  );
+
+  return (
+    <Column>
+      <InputField.Root  id={'colorName'}>
+          <InputField.Input 
+            value   = {name === "New Color Variable" || name === "Multiple" ? '' : name} 
+            placeholder = {name}
+            onChange = {onInputChange} 
+          />
+      </InputField.Root >
+      <ColorInputFieldFull
+            id={colorInputId}
+            value={color}
+            onChange={onChangeColor}
+      />
+      <Row id={id}>
+        <LabeledElementView renderLabel={renderLabel}>
+          <Spacer.Vertical size={8} />
+          <InputField.Root id={hexInputId} labelPosition="start">
+            <InputField.Input value={hexValue} onSubmit={() => {}} />
+            <InputField.Label>#</InputField.Label>
+          </InputField.Root>
+          <Spacer.Horizontal size={8} />
+          <InputField.Root id={opacityInputId} size={50}>
+            <InputField.NumberInput
+              value={Math.round(color.alpha * 100)}
+              onSubmit={handleSubmitOpacity}
+              onNudge={handleNudgeOpacity}
+            />
+            <InputField.Label>%</InputField.Label>
+          </InputField.Root>
+        </LabeledElementView>
+      </Row>
+    </Column>
+  );
+});
+
+
+const ColorPickerInspector =  memo(function ColorPickerInspector({swatches, index}: ColorProps) {
+  const [, dispatch] = useApplicationState();
+  const swatch: Sketch.Swatch = swatches[0];
+
+  const color = swatch.value as Sketch.Color;
+  const rgbObj = {
+    "r": Math.round(color.red   * 255),
+    "g": Math.round(color.green * 255),
+    "b": Math.round(color.blue  * 255)
+  }
+
+  function rgb2hex(rgb: string){
```

Comment:
Is there an existing function we can use from https://github.com/noya-app/noya/blob/6af9eb378e9a26cdb5d241184864249991819c25/packages/noya-colorpicker/src/utils/convert.ts?

Reply:
Now using the existing function